src/pages/business-page/business-page.js

```diff
@@ -18,14 +22,85 @@ const BusinessPage = ({ match }) => {
     onCompleted: (data) => setPage(data.getBusinessTextByCode)
   });
 
-  const addressText = page?.text && parse(t(`${page.translationsKey}.text`));
+  const text = page?.text && t(`${page.translationsKey}.text`);
+  const addressText = text && parse(text);
+
+  useEffect(() => {
+    setIndexImg([]);
+    if (addressText) {
+      setTitle(addressText[0].props.children);
+      addressText.map((item, index) => {
+        if (item.type === 'img') {
+          setIndexImg((prev) => [...prev, index]);
+        }
+        return null;
+      });
+    }
+  }, [page, text]);
+
+  useEffect(() => {
+    setResult([]);
```

Comment:
I don't think we need to change state twice in this useEffect, make a dependency with which it will choose a value for it.

Reply:
?